### ISSUE TITLE
Missing strings

### ISSUE THREAD

**Issue by morphinestyle** [akeeba/panopticon#352]
Missing strings in the site panel.

![IMG_20231201_014745.jpg](https://github.com/akeeba/panopticon/assets/35969533/bd824184-8856-4b52-ab30-0a8faa8f8e05)

**Comment by nikosdion** [comment]
Please see https://github.com/akeeba/panopticon/discussions/353

**Comment by brianteeman** [comment]
The problem is that the german translation file only has 229 lines instead of 1695 lines

**Comment by morphinestyle** [comment]
Strange situation: I just searched on Weblate for one of the apparently missing strings PANOPTICON_SITES_LBL_DROPDOWN_AUTOMATIONS_HEAD_BACKUP_SECURITY: in English Weblate shows it to me, it's present, while in Italian it doesn't find it, it's missing...

Shouldn't Weblate show all the strings in English in other languages ​​too?

**Comment by nikosdion** [comment]
It _should_ show new strings. That is the entire reason of using something like Weblate. But you are right, it doesn't. Some language strings it just… ignores? They are different on every language as well.

**Comment by morphinestyle** [comment]
> The problem is that the german translation file only has 229 lines instead of 1695 lines

Same situation: English has 1,605 strings, while Italian has 1,592.
What am I doing wrong?

**Comment by nikosdion** [comment]
I don't think any of us is doing anything wrong. Something is not right with Weblate. This is really unfortunate, because this is the only affordable translation environment. The pricing of other translation services (e.g. CrowdIn, Transifex, etc) is unrealistic. They also won't give a free account to small projects like ours. 

I could write scripts to convert between GNU GetText POT and INI files and have you use Poedit to translate, then do PRs with the translations. The problem is that this is developer-friendly, not translator-friendly. I was trying to remove that barrier to entry, and also the fact that in cases like that I end up spending all my time managing translation files instead of working on code (the overlap of people with the base skills to work with GitHub and the people who are willing to do translations is somewhere between scant to non-existent).

I don't know what is the solution, or if one even exists to be honest.

**Comment by brianteeman** [comment]
In another project I am involved with they use [poedit](https://poeditor.com/pricing/) which is free for open source projects and it directly connects with github and supports ini files

**Comment by nikosdion** [comment]
@brianteeman It supports INI files now? Well, that's one less step in what I described (as you saw, I did mention Poedit above). 

The problem that the translation has a higher barrier to entry remains, though. I would only be able to accept translations as PRs. You and I can do that, but can _all_ translators do it? In some other free software I have asked volunteer translators to do that and all I got was file attached in the issue comments. That's okay for once a year updates, but it's not manageable for multiple times per week, per language updates. I want to spend my time writing code, not managing files.

If anyone would volunteer to be the translations manager, helping people set up GitHub Desktop and poedit, use them to translate and send PRs, I would be more than happy to document the process and start asking people to use that process for translating instead. Any takers? Pretty please? 😉

**Comment by brianteeman** [comment]
It does the pr automatically. Translators dont need to do anything

**Comment by nikosdion** [comment]
@brianteeman Thank you very much for the very interesting feedback! I will play with it and see if I can come up with a decent, documented translation process.

**Comment by nikosdion** [comment]
Ugh, I don't like that. It requires an action to pull the translations via an API. This actually costs me money, because _GitHub_ doesn't give me free action minutes; these come from my own pocket.

Also, I was talking about the standalone software [Poedit](https://poedit.net/). I was not talking about a service. Weblate is also free for FOSS projects, but look where we are now, still wondering why we have problems with translations.

**Comment by Rolli1962** [comment]
In case of giving weblate a chance...
Have a little bit compared. Seems it do only create entries in the ini-files, if a translation is given.
Means if it would be possible in the code of panopticon to output the translated words only, if the language-string is set in the ini-file of the language. If not to use the default-language (English) for those output.
To be honest - I do not know how much effort it takes to implement it as code.

**Comment by nikosdion** [comment]
You are confusing two different things.

Partial translations is one thing. This should not be happening, and I need to see why it's happening for some people (when I tried the same locally I did get the untranslated strings in English, as it should be the case). That's not a very big issue, especially since version 1.0.5 lets you select a specific language in your user profile.

Weblate randomly not even giving us the option to translate strings in some languages – different strings, in different languages – is a different thing. I cannot address it for them. It's literally its one job. Not having a reliable translation environment is about as useful as having no translation environment. So, that's what I want to address next.
